### PR TITLE
API documentation

### PR DESCRIPTION
**Pull request by imranashraf**
automatic documentation generation for API
removed deprecated doc folder
notebooks are available in examples/notebooks directory
discussed notebooks, examples and tests in getting started page

### CODE REVIEW

**Comment by AdriaanRol** [comment]
@imranashraf , I tried to review this documentation but I cannot build it locally because I did not succeed in installing OpenQL from the repository. Somehow the `pip install qutechopenql` is not recognized when building this documentation. 

A solution (for all doc reviewing) would be to enable the branch in the admin interface of `readthedocs`. that way it is easier for me (and others) to quickly check out the work.

File: docs/start.rst [hunk 1]
```diff
@@ -67,4 +73,38 @@ A good place to get started with with your own programs might be to copy `exampl
...
 
-    discuss the generated output files
\ No newline at end of file
+    discuss the generated output files
+
+
+Notebooks
+---------
+
+Following Jupyter notebooks are available in ``<OpenQL Root Dir>/examples/notebooks`` directory:
```
Comment (AdriaanRol) [inline docs/start.rst]: notebooks are available in *the* ``<OpenQL Root Dir>/examples/notebooks`` directory

Add "the"  in multiple places

Comment (imranashraf) [reply docs/start.rst]: fixed

**Comment by imranashraf** [comment]
@AdriaanRol the automatic built setup has been updated to build api docs on RTD.

**Pull request merged**